core/server/master/src/main/java/alluxio/master/metrics/DefaultMetricsMaster.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.metrics;
+
+import alluxio.Constants;
+import alluxio.clock.SystemClock;
+import alluxio.master.AbstractMaster;
+import alluxio.master.MasterContext;
+import alluxio.metrics.Metric;
+import alluxio.metrics.MetricsAggregator;
+import alluxio.metrics.MetricsFilter;
+import alluxio.metrics.MetricsStore;
+import alluxio.metrics.MetricsSystem;
+import alluxio.metrics.WorkerMetrics;
+import alluxio.metrics.aggregator.SumInstancesAggregator;
+import alluxio.proto.journal.Journal.JournalEntry;
+import alluxio.util.executor.ExecutorServiceFactories;
+import alluxio.util.executor.ExecutorServiceFactory;
+
+import com.codahale.metrics.Gauge;
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.thrift.TProcessor;
+
+import java.io.IOException;
+import java.time.Clock;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Default implementation of the metrics master.
+ */
+public class DefaultMetricsMaster extends AbstractMaster implements MetricsMaster {
+  private final Map<String, MetricsAggregator> mMetricsAggregatorRegistry = new HashMap<>();
+  private final MetricsStore mMetricsStore;
+
+  /**
+   * Creates a new instance of {@link MetricsMaster}.
+   *
+   * @param masterContext the context for metrics master
+   */
+  protected DefaultMetricsMaster(MasterContext masterContext) {
+    this(masterContext, new SystemClock(), ExecutorServiceFactories
+        .fixedThreadPoolExecutorServiceFactory(Constants.METRICS_MASTER_NAME, 2));
+  }
+
+  /**
+   * Creates a new instance of {@link MetricsMaster}.
+   *
+   * @param masterContext the context for metrics master
+   * @param clock the clock to use for determining the time
+   * @param executorServiceFactory a factory for creating the executor service to use for running
+   *        maintenance threads
+   */
+  protected DefaultMetricsMaster(MasterContext masterContext, Clock clock,
+      ExecutorServiceFactory executorServiceFactory) {
+    super(masterContext, clock, executorServiceFactory);
+    mMetricsStore = masterContext.getMetricsStore();
```

Comment:
Would it make sense for the `MetricsMaster` to own the `mMetricsStore`? Other masters can take a dependency on `MetricsMaster` and access the store through `MetricsMaster`. It's odd how there's a `MetricsMaster` but it doesn't own the `MetricsStore`

Reply:
Thinking it twice, passing metric master to block master makes more sense.